components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQMessageConverter.java

```diff
@@ -203,18 +202,91 @@ private Object getValidRabbitMQHeaderValue(Object headerValue) {
             return headerValue;
         } else if (headerValue instanceof LongString) {
             return headerValue;
-        } else if (headerValue instanceof Timestamp) {
-            return headerValue;
-        } else if (headerValue instanceof Byte) {
-            return headerValue;
-        } else if (headerValue instanceof Double) {
-            return headerValue;
-        } else if (headerValue instanceof Float) {
-            return headerValue;
-        } else if (headerValue instanceof Long) {
-            return headerValue;
         }
```

Comment:
BigDecimal, Byte, Double, Float and Long are all Numbers so no need to explicitly check them too; Timestamp is a java.util.Date so already covered